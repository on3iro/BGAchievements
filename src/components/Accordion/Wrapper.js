import styled from 'styled-components'

const Wrapper = styled.div`
  width: 100%;
  height: 90%;

  overflow: auto;

  div:last-child {
    margin-bottom: 0;
  }
`

export default Wrapper
