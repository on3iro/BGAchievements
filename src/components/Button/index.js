/**
 * A common button, if you pass it a prop "route" it'll render a link to a react-router route.
 * Otherwise it'll render a link with an onclick.
 *
 * @namespace Button
  */

import React, { Children } from 'react'
import PropTypes from 'prop-types'

import A from './A'
import StyledSubmit from './StyledSubmit'
import StyledNavLink from './StyledNavLink'

const Button = props => {
  // Render an anchor tag
  let button = (
    <A href={props.href} onClick={props.onClick} {...props} >
      {Children.toArray(props.children)}
    </A>
  )

  // If the button has a 'to' prop, we want to render a NavLink
  if (props.to) {
    button = (
      <StyledNavLink {...props} >
        {Children.toArray(props.children)}
      </StyledNavLink>
    )
  }

  // If the button has submit prop render input
  if (props.submit) {
    button = (
      <StyledSubmit {...props} />
    )
  }

  return button
}

Button.propTypes = {
  handleRoute: PropTypes.func,
  href: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node,
  submit: PropTypes.bool
}

export default Button
