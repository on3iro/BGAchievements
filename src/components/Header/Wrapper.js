/**
  * Styled container for Header
  *
  *
  * @namespace Header.Wrapper
  * @memberOf Header
  */

import styled from 'styled-components'

const Wrapper = styled.header`
  position: fixed;
  top: 0px;
  margin-bottom: 0;

  min-height: 64px;
  height: 64px;
  width: 100%;

  background-color: ${props => props.theme.mainBG};
  color: ${props => props.theme.noneWhite};

  @media (max-width: 900px) {
    position: static;

    height: auto;
  }
`

export default Wrapper
