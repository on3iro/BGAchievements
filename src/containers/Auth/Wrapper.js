/**
  * Renders a styled div for the PasswordGenerator
  *
  * @namespace PasswordGenerator.Wrapper
  * @memberOf PasswordGenerator
  */

import styled from 'styled-components'

const Wrapper = styled.section`
  margin-top: 20px;
`

export default Wrapper
