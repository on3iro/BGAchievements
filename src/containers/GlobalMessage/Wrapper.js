import styled from 'styled-components'

const Wrapper = styled.div`
  margin: 80px 0 0 0;
  padding: 10px 20px;
  height: 40px;

  border-radius: 4px;

  background: ${props => props.error ? props.theme.warningHover : props.theme.primaryHover};
  opacity: ${props => props.empty ? '0' : '.6'};
  color: #fff;
`

export default Wrapper
