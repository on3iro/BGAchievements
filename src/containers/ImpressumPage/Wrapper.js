import styled from 'styled-components'

const Wrapper = styled.section`
  width: 100%;
  height: 100%;

  overflow: auto;
`

export default Wrapper
