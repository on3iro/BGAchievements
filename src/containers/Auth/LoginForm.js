/**
  * Renders a basic LoginForm
  *
  * @namespace PasswordGenerator.LoginForm
  * @memberOf PasswordGenerator
  */

import React from 'react'
import PropTypes from 'prop-types'
import { Field, reduxForm } from 'redux-form'

import H1 from 'components/H1'
import Button from 'components/Button'
import LoadingSpinner from 'components/LoadingSpinner'

import FormField from 'components/FormField'

export const validate = values => {
  const errors = {}
  const REQUIRED = 'Pflichtfeld'

  if (!values.email) {
    errors.email = REQUIRED
  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
    errors.email = 'UngÃ¼ltige Email-Adresse!'
  }

  if (!values.password) {
    errors.password = REQUIRED
  }

  return errors
}

export const LoginForm = props => {
  return (
    <form onSubmit={props.handleSubmit}>
      <H1>Login</H1>
      <Field name='email' type='text'
        component={FormField} label='Email'
      />
      <Field name='password' type='password'
        component={FormField} label='Passwort'
      />
      {
        props.isLoading
        ? <Button onClick={(e) => e.preventDefault}>
          <LoadingSpinner controls />
        </Button>
        : <Button submit >Anmelden</Button>
      }
    </form>
  )
}

LoginForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  isLoading: PropTypes.bool
}

// Decorate the form component
export default reduxForm({
  form: 'login',
  validate
})(LoginForm)
