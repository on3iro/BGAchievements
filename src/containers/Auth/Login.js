import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import Button from 'components/Button'

import LoginForm from './LoginForm'
import FormDivider from './FormDivider'
import Wrapper from './Wrapper'

import { loginSubmit } from './ducks/actions'
import { getLoadingState } from './ducks/selectors'

export class Login extends Component {
  handleSubmit = values => {
    this.props.loginSubmit(values)
  }

  render () {
    return (
      <Wrapper>
        <LoginForm isLoading={this.props.isLoading} onSubmit={this.handleSubmit} />
        <FormDivider>Noch keinen Account?</FormDivider>
        <Button to='/register' secondary>Registrieren</Button>
      </Wrapper>
    )
  }
}

Login.propTypes = {
  isLoading: PropTypes.bool,
  loginSubmit: PropTypes.func.isRequired
}

export const mapStateToProps = state => {
  return {
    isLoading: getLoadingState(state)
  }
}

export const mapDispatchToProps = dispatch => {
  return bindActionCreators({
    loginSubmit
  }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Login)
