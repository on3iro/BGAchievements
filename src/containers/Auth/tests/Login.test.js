import React from 'react'
import { shallow } from 'enzyme'

import Button from 'components/Button'

import Wrapper from '../Wrapper'
import { Login, mapStateToProps, mapDispatchToProps } from '../Login'
import LoginForm from '../LoginForm'

const renderComponent = newProps => {
  const props = {
    loginSubmit: jest.fn(),
    ...newProps
  }

  const renderedWrapper = shallow(<Login {...props} />)

  return {
    props,
    renderedWrapper
  }
}

describe('<Login />', () => {
  it('should render <Wrapper />', () => {
    const { renderedWrapper } = renderComponent()
    expect(renderedWrapper.find(Wrapper).length).toMatchSnapshot()
  })

  it('should render <LoginForm />', () => {
    const { renderedWrapper } = renderComponent()
    expect(renderedWrapper.find(LoginForm).length).toMatchSnapshot()
  })

  it('should render <Button />', () => {
    const { renderedWrapper } = renderComponent()
    expect(renderedWrapper.find(Button).length).toMatchSnapshot()
  })

  it('should call loginSubmit() on handleSubmit', () => {
    const { renderedWrapper, props } = renderComponent()

    renderedWrapper.instance().handleSubmit('test')

    expect(props.loginSubmit).toHaveBeenCalledWith('test')
  })

  describe('react-redux', () => {
    it('should map state to props', () => {
      expect(mapStateToProps({
        auth: { AuthRequestState: { isLoading: true }}
      })).toMatchSnapshot()
    })

    it('should map dispatch to props', () => {
      expect(mapDispatchToProps(() => null)).toMatchSnapshot()
    })
  })
})
