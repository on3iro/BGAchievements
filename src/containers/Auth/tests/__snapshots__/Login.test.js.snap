exports[`<Login /> react-redux should map dispatch to props 1`] = `
Object {
  "loginSubmit": [Function],
}
`

exports[`<Login /> react-redux should map state to props 1`] = `
Object {
  "isLoading": true,
}
`

exports[`<Login /> should render <Button /> 1`] = `1`

exports[`<Login /> should render <LoginForm /> 1`] = `1`

exports[`<Login /> should render <Wrapper /> 1`] = `1`
